build.gradle

```diff
@@ -118,11 +126,51 @@ dependencies {
     testCompile 'org.powermock:powermock-api-mockito2:2.+'
     
     //WA Libraries
-    implementation 'org.wycliffeassociates:8woc2018-common'
-    implementation 'org.wycliffeassociates:kotlin-resource-container'
-    implementation 'com.github.WycliffeAssociates:jdenticon-kotlin:-SNAPSHOT'
+    api 'org.wycliffeassociates:8woc2018-common'
+    api 'org.wycliffeassociates:kotlin-resource-container'
+    api 'com.github.WycliffeAssociates:jdenticon-kotlin:-SNAPSHOT'
```

Comment:
`If the types from a dependency appear in public fields or methods of your library’s public classes, then that dependency is exposed via your library’s public API and should therefore be added to the api configuration. Otherwise, the dependency is an internal implementation detail and should be added to implementation.` - Gradle Documentation.

I don't know enough about the codebase to speak intelligently to this. Thoughts? It kind of sounds like we might be our own consumer and therefore would need API.

Reply:
Further reference: https://docs.gradle.org/current/userguide/java_library_plugin.html#sec:java_library_recognizing_dependencies

Reply:
I tested this change and things seem to be working. I thought I tried this already and it wouldn't run, but maybe something we changed elsewhere fixed the issue that I thought was caused by using `implementation` here. If you read through the documentation and think that we're not using any of the things that would need API I'm good with changing to implementation.

Reply:
**TL;DR: let's go with `implementation` in the jvm repo**

I think the issue here is whether or not we are developing a library. So in this case JVM is not a library, so we can use implementation.

As a contrary example, 8woc2018-common *is* a library (we are depending on it), and it exposes public APIs (e.g., `ImportResourceContainer`) which use data types from RxKotlin. So it would make sense to specify the rx dependencies in common with `api`. The advantage of this is that if I want to use the common library in another project (like the JVM repo) I don't have to add a separate dependency for RxKotlin -- the appropriate RxKotlin version needed for common is already exposed through common itself. If we used `implementation` for the rx dependencies in common, we would be required to have a dependency on rx directly in JVM in order to have the data types needed to interact with the common library. 

This has not raised its head so far since we needed (and so included) RxKotlin for the repo code in JVM anyway, so it hasn't mattered whether the rx api was exposed from common or not.

Reply:
Sounds good.

Reply:
If I'm understanding this correctly, then I think what we *could* do is modify the build.gradle for common, have kotlin-resource-container there use API, and then have everything here be implementation?

Seems to me we landed on api though because class definitions weren't getting packaged into the jar though.

Reply:
I think they are getting packaged even with implementation due to the `runtimeClasspath` change, now.